app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt

```diff
@@ -119,16 +119,18 @@ class ContactDetailsFragment : Fragment() {
         }
 
         view.findViewById<Button>(R.id.run_button).setOnClickListener {
-            val route = DUMMY_ROUTE[0] //remove when merged with contact creation and use real route
             val uri =
-                Uri.parse("google.navigation:q=${route.dst.latitude},${route.dst.longitude}&mode=b")
+                Uri.parse("google.navigation:q=${currentContact!!.latitude},${currentContact!!.longitude}&mode=b")
```

Comment:
It would be better to change the currentContact class attribute to not nullable, and handle the null case at the initialisation in onViewCreated(), than to call everywhere .!!. It would add more readability and comprehension for that class.